### PR TITLE
ci: build and deploy pr preview

### PR DESCRIPTION
**Pull request by ivan-ngchakming**
## Changes
- added `.vscode` to .gitignore and removed related files
- added setup instructions to readme
- added ci workflow to detect changes in `./frontend` and deploy to Netlify as preview site if changes is detected.

~~Blocked by:~~
~~Need to add `NETLIFY_AUTH_TOKEN` and `NETLIFY_SITE_ID` to repo secrets.~~
~~I don't have permission to do that, so we can either~~
~~1. create a github organization and transfer this repo to that org, so I can have access to repo settings, or~~
~~2. @toonvanvr you create a netlify project using your github account, and fill in the repo secrets~~

### CODE REVIEW

**Comment by toonvanvr** [comment]
@ivan-ngchakming 
I'd like to try the Github Organization approach. Who should take care of that? You seem very organized, so I don't mind if you're "owner" of it or whatever roles they provide.

File: .gitignore [hunk 1]
```diff
@@ -1 +1,4 @@
 node_modules/
+
+# IDE
+**/.vscode
```
Comment (toonvanvr) [inline .gitignore:4]: The default angular template does not fully gitignore the .vscode folder. Can we keep files shared, like the file with "pluginrecommendations" etc? 

```suggestion
# IDEs and editors
.idea/
.project
.classpath
.c9/
*.launch
.settings/
*.sublime-workspace

# Visual Studio Code
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json
.history/*
```

Comment (ivan-ngchakming) [reply .gitignore]: [question] what are these files used for? I thought files under .vscode is used to configure a personalized IDE workspace. Also do we need to move this folder to root of the project for it to work?

**Comment by ivan-ngchakming** [comment]
Created organization and sent you an invite
https://github.com/geo-event-tracker

File: .gitignore (hunk 1 above)
Comment (toonvanvr) [reply .gitignore]: Hmm, indeed, nevermind what I proposed, it was a bad suggestion. I'll take a look at what _can_ be shared from the `.vscode` folder in a separate feature at some point. For now we can indeed just gitignore it all.

File: frontend/README.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # EventTrackerFrontend
 
+:)
```
Comment (toonvanvr) [inline frontend/README.md:3]: lmao

**Review by toonvanvr — APPROVED** [review]
Looks good!

What's your reason to not include sourcemaps in the CI build, @ivan-ngchakming? Not blocking, just curious.

**Comment by ivan-ngchakming** [comment]
> Looks good!
> 
> What's your reason to not include sourcemaps in the CI build, @ivan-ngchakming? Not blocking, just curious.

Reduce bundle size I guess (the flag is there coz I copied it online lol), I was planning to use the same workflow and build command for deploying the production build. Can add it back later if needed, but for pr previews I guess we'll mostly use it to see visual changes and do UATs, thus less need for source map.

**Pull request merged**